components/public-api-server/pkg/apiv1/oidc.go

```diff
@@ -350,3 +354,39 @@ func assertIssuerIsReachable(host string) error {
 	}
 	return nil
 }
+
+func assertIssuerProvidesDiscovery(issuer string) error {
+	tr := &http.Transport{
+		TLSClientConfig: &tls.Config{InsecureSkipVerify: true},
+		Proxy:           http.ProxyFromEnvironment,
+	}
+	client := &http.Client{
+		Transport: tr,
+		Timeout:   2 * time.Second,
+		// never follow redirects
+		CheckRedirect: func(*http.Request, []*http.Request) error {
+			return http.ErrUseLastResponse
+		},
+	}
+
+	resp, err := client.Get(issuer + "/.well-known/openid-configuration")
+	if err != nil {
+		return err
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode != 200 {
+		return fmt.Errorf("OIDC Discovery is not provided.")
+	}
+
+	var config map[string]interface{}
+	err = json.NewDecoder(resp.Body).Decode(&config)
```

Comment:
Should ideally check the `Content-Type` of the response is JSON before attempting JSON decode

Reply:
I was not sure if all IdPs would sent the content type properly. But there is no real reason to not check for the it. 

👍🏻